docs/pages/index.md

```diff
@@ -0,0 +1,51 @@
+# SKNNR Docs
+
+`scikit-learn-knn-regression` (a.k.a. `sknnr`) is a package for running k-nearest neighbor imputation methods, including GNN (Ohmann & Gregory, 2002), MSN (Moeur & Stage 1995), and RFNN (Crookston & Finley, 2008), using estimators that are fully compatible with [`scikit-learn`](https://scikit-learn.org/stable/).
+
```

Comment:
I sometimes include a `Features` section here in my projects with a bulleted list of the main selling points of the package. Might be worth doing here, although I wasn't quite sure what all to include.

Reply:
I totally agree.  I looked at the format of your `wxee` repo and I like the headings you have there. e.g.

* What is &lt;package&gt;?
* Features
* (In the case of `sknnr`, perhaps a longer description of the history of the research along with common approaches)

Again, I'll want to think a bit about the specific features that we have for this section.

Reply:
I got totally high-centered on "features".  You allude to some of these in the "Examples" section (independent scoring, dataframe support, the additional y-fit matrix, and dimensionality reduction).  I found myself reiterating these, but felt that it would be better left to the examples.

If we think of features as why would someone use this package versus something else, we may be able to claim:

* tight integration with scikit-learn API
* support for dataframes
* native multi-output support
* others?

Reply:
I agree that we should stick with these high-level features up front and leave the lower level details for the Usage guide. Your list looks pretty complete to me, and I think those are all good selling points. I suppose we could also point out that estimators can be used for regression and classification (with k=1), since that's kind of unique (although you do already mention that in the name explanation). The only other thing I can think of to mention is that results are verified against `yaImpute`, but maybe we don't want to tie ourselves to that if we eventually add functionality we can't test with `yaImpute`?

Reply:
OK, let's go ahead and include these in a "Features" section (are you able to do the honors?).  For consistency with `yaImpute`, I feel like that is a selling point for users who may come from an R background so I'm in favor of including it.  The only thing that gives me pause is being able to completely reproduce `RFNN` because we're at the mercy of two different implementations of random forests and if those don't agree, we'll have some issues.  I think we can say we strive for complete verification against `yaImpute` when possible/practical.

Reply:
Done! I didn't add a disclaimer about `RFNN` validation yet, but I think we can just tack that onto the footnote later if needed - getting different results with different RF implementations seems totally reasonable to me. (While I was thinking about that, I also added another footnote just to note that `RFNN` is still in development).

Do you think we should have a short section in the Usage guide on the regression vs. classification modes? You already covered the main points in the name explanation, but it might be nice to have it in both places since it seems like an important distinction.

Reply:
Oh man, those emojis are adorable!  I really like them :smile:  This is really looking great!

> Do you think we should have a short section in the Usage guide on the regression vs. classification modes? You already covered the main points in the name explanation, but it might be nice to have it in both places since it seems like an important distinction

I think that sounds like a good idea.  We don't necessarily have the data to support it, but it would be cool to show how you might "stitch" together the results of continuous attributes (features) and classified attributes (features) in order to get a predicted file that had that hybrid.  We'll need to do that in however we handle accuracy assessment (outside of `sknnr`) but for practitioners who are familiar with kNN mapping, this is a pretty common use pattern.

Please let me know if I can help with this in any way.

Reply:
I added this section to the Usage guide, but I did start wondering partway through whether it was really worth including since it's just the standard functionality of `KNeighborsRegressor`. Let me know what you think, and feel free to edit if you have any changes!

> it would be cool to show how you might "stitch" together the results of continuous attributes (features) and classified attributes (features) in order to get a predicted file that had that hybrid

Agreed! I thought that was probably too ambitious to include here as a code block, but I have been thinking that we should add a few Jupyter notebooks to the docs (using [mkdocs-jupyter](https://github.com/danielfrg/mkdocs-jupyter)) for more detailed examples that could be run interactively with Binder or Colab, and this seems like a great applied example. In the interest of wrapping this up, I'm thinking this would probably be a separate PR.

Reply:
> I added this section to the Usage guide, but I did start wondering partway through whether it was really worth including since it's just the standard functionality of `KNeighborsRegressor`. Let me know what you think, and feel free to edit if you have any changes!

Even though it's the standard functionality of `KNeighborsRegressor`, I think the way you've written it provides some additional clarity that users may not have thought about when using `KNeighborsRegressor` with `n_neighbors` = 1.  So I really like the additional information here.  A few optional edits that are very much nits, so you can take or leave as you want:

* I first stumbled when reading `n_neighbors>1` (all in markdown code).  To my eye, `n_neighbors` > 1 and `n_neighbors` = 1 are easier to read.
* Because the default weighting option is `uniform`, users might be surprised that predictions would come from weighted averages.  I think we can qualify this by saying "optionally-weighted averages" or "(optionally-) weighted averages".
* Possibly change "where each attribute is imputed directly from another plot" to "where each attribute is imputed directly from its nearest neighbor".

Reply:
Good calls all around! Just made those changes.